### PR TITLE
Remove client secret usage in pipelines

### PR DESCRIPTION
**Pull request by yophilav**
- Remove the client secret
- Workaround the DevOps task where az cli cannot authorize itself within bash script in AzCLI task

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: scripts/linux/nested-edge-deploy-agent.sh [hunk 1]
```diff
@@ -123,7 +123,7 @@ function prepare_test_from_artifacts() {
     sudo cat ${deployment_working_file}
 
     #deploy the config in azure portal
-    az iot edge set-modules --device-id ${DEVICE_ID} --hub-name ${IOT_HUB_NAME} --content ${deployment_working_file} --output none
+    #az iot edge set-modules --device-id ${DEVICE_ID} --hub-name ${IOT_HUB_NAME} --content ${deployment_working_file} --output none
```
Comment (marianan) [inline scripts/linux/nested-edge-deploy-agent.sh:126]: should this line be removed all together?

File: scripts/linux/nested-edge-deploy-agent.sh [hunk 2]
```diff
@@ -179,12 +179,12 @@ function process_args() {
         elif [ $saveNextArg -eq 16 ]; then
             CONNECTION_STRING="$arg"
             saveNextArg=0
-        elif [ $saveNextArg -eq 17 ]; then
-            DEVICE_ID="$arg"
-            saveNextArg=0
-        elif [ $saveNextArg -eq 18 ]; then
-            IOT_HUB_NAME="$arg"
-            saveNextArg=0
+        # elif [ $saveNextArg -eq 17 ]; then
```
Comment (marianan) [inline scripts/linux/nested-edge-deploy-agent.sh:182]: same as above, if not needed, maybe remove all together or add a comment

File: scripts/linux/nested-edge-deploy-agent.sh [hunk 3]
```diff
@@ -226,7 +226,7 @@ function process_args() {
     done
 
     # Required parameters
-    [[ -z "$DEVICE_ID" ]] && { print_error 'DEVICE_ID is required.'; exit 1; }
+    # [[ -z "$DEVICE_ID" ]] && { print_error 'DEVICE_ID is required.'; exit 1; }
```
Comment (marianan) [inline scripts/linux/nested-edge-deploy-agent.sh:229]: same as above, if not needed, maybe remove all together or add a comment

**Comment by yophilav** [comment]
Updated.

**Review by marianan — APPROVED** [review]

**Review by damonbarry — APPROVED** [review]

**Pull request merged**